lib/block_view_stake.go

```diff
@@ -0,0 +1,2496 @@
+		LockedAmountNanos:   lockedStakeEntry.LockedAmountNanos.Clone(),
+		LockedAtEpochNumber: lockedStakeEntry.LockedAtEpochNumber,
+		ExtraData:           copyExtraData(lockedStakeEntry.ExtraData),
+		isDeleted:           lockedStakeEntry.isDeleted,
+	}
+}
+
+func (lockedStakeEntry *LockedStakeEntry) Eq(other *LockedStakeEntry) bool {
+	return lockedStakeEntry.LockedStakeID.IsEqual(other.LockedStakeID)
+}
+
+func (lockedStakeEntry *LockedStakeEntry) ToMapKey() LockedStakeMapKey {
+	return LockedStakeMapKey{
+		StakerPKID:          *lockedStakeEntry.StakerPKID,
+		ValidatorPKID:       *lockedStakeEntry.ValidatorPKID,
+		LockedAtEpochNumber: lockedStakeEntry.LockedAtEpochNumber,
+	}
+}
+
+func (lockedStakeEntry *LockedStakeEntry) RawEncodeWithoutMetadata(blockHeight uint64, skipMetadata ...bool) []byte {
+	var data []byte
+	data = append(data, EncodeToBytes(blockHeight, lockedStakeEntry.LockedStakeID, skipMetadata...)...)
+	data = append(data, EncodeToBytes(blockHeight, lockedStakeEntry.StakerPKID, skipMetadata...)...)
+	data = append(data, EncodeToBytes(blockHeight, lockedStakeEntry.ValidatorPKID, skipMetadata...)...)
+	data = append(data, EncodeUint256(lockedStakeEntry.LockedAmountNanos)...)
+	data = append(data, UintToBuf(lockedStakeEntry.LockedAtEpochNumber)...)
+	data = append(data, EncodeExtraData(lockedStakeEntry.ExtraData)...)
+	return data
+}
+
+func (lockedStakeEntry *LockedStakeEntry) RawDecodeWithoutMetadata(blockHeight uint64, rr *bytes.Reader) error {
+	var err error
+
+	// LockedStakeID
+	lockedStakeID := &BlockHash{}
+	if exist, err := DecodeFromBytes(lockedStakeID, rr); exist && err == nil {
+		lockedStakeEntry.LockedStakeID = lockedStakeID
+	} else if err != nil {
+		return errors.Wrapf(err, "LockedStakeEntry.Decode: Problem reading LockedStakeID: ")
+	}
+
+	// StakerPKID
+	stakerPKID := &PKID{}
+	if exist, err := DecodeFromBytes(stakerPKID, rr); exist && err == nil {
+		lockedStakeEntry.StakerPKID = stakerPKID
+	} else if err != nil {
+		return errors.Wrapf(err, "LockedStakeEntry.Decode: Problem reading StakerPKID: ")
+	}
+
+	// ValidatorPKID
+	validatorPKID := &PKID{}
+	if exist, err := DecodeFromBytes(validatorPKID, rr); exist && err == nil {
+		lockedStakeEntry.ValidatorPKID = validatorPKID
+	} else if err != nil {
+		return errors.Wrapf(err, "LockedStakeEntry.Decode: Problem reading ValidatorPKID: ")
+	}
+
+	// LockedAmountNanos
+	lockedStakeEntry.LockedAmountNanos, err = DecodeUint256(rr)
+	if err != nil {
+		return errors.Wrapf(err, "LockedStakeEntry.Decode: Problem reading LockedAmountNanos: ")
+	}
+
+	// LockedAtEpochNumber
+	lockedStakeEntry.LockedAtEpochNumber, err = ReadUvarint(rr)
+	if err != nil {
+		return errors.Wrapf(err, "LockedStakeEntry.Decode: Problem reading LockedAtEpochNumber: ")
+	}
+
+	// ExtraData
+	lockedStakeEntry.ExtraData, err = DecodeExtraData(rr)
+	if err != nil {
+		return errors.Wrapf(err, "LockedStakeEntry.Decode: Problem reading ExtraData: ")
+	}
```

Comment:
see PR here to use DecodeDeSoEncoder: https://github.com/deso-protocol/core/pull/505